Assets/Code/Scripts/Level generation scripts/TerrainGeneratorScript.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class TerrainGeneratorScript : MonoBehaviour
+{
+    private float tileWidth = 10.0f;
+
+    public int blocks = 2; //number of blocks being made, block x block
+    public int size = 5;
+    public float tavernPercent = 0.25f;
+    private RoadGenerationScript _roadScript;
+    private CrossGenerationScript _crossRoadScript;
+    private TavernGenerationScript _tavernScript;
+    private FillerGenerationScript _fillerScript;
```

Comment:
I've noticed that every script here overrides the parent function provided. Also, we only need a single instance of everything. Instead, what do you guys think of making these classes static, with a data struct similar to the score to hold the `size`, `width`, and `block`? This would allow us to reduce the number of parameters in this script as well as we could directly call the static functions.

Reply:
static classes can't Instantiate game objects. 
also it is not overwriting. An abstract is designed to make a set of functions that i want to guarantee are in the class that follows it's pattern. This will come in handy if in the future when the blocks become more elaborate, we can add any necessary method abstracts.
This way once we discover something that we need to add, we add it to the parent and the code won't run unless the children are updated.